src/database/memory.rs

```diff
@@ -449,6 +490,54 @@ impl Database for MemoryDatabase {
 
         Ok(*value)
     }
+
+    fn del_spent_utxos(&mut self, spent_utxos: Vec<LocalUtxo>) -> Result<Vec<LocalUtxo>, Error> {
+        let mut deleted_utxos: Vec<LocalUtxo> = vec![];
+        if spent_utxos.is_empty() {
+            let all_spent_utxos: Vec<LocalUtxo> = self.select_spent_utxos();
+            self.delete_spent_utxo_set(&mut deleted_utxos, &all_spent_utxos)?;
+        } else {
+            self.delete_spent_utxo_set(&mut deleted_utxos, &spent_utxos)?;
+        }
+
+        Ok(deleted_utxos)
```

Comment:
if you don't pass in the deleted utxo collector this could would simply look like 
```
if spent_utxos.is_empty() {
           Ok(self.delete_spent_utxo_set(&self.select_spent_utxos())?);
        } else {
            Ok(self.delete_spent_utxo_set(&spent_utxos)?);
        }
```